src/ctypes-foreign-base/dl.ml.win

```diff
@@ -82,7 +84,7 @@ let dlopen ?filename ~flags =
           else
             x
         in
-        let ls = String.lowercase s in
+        let ls = String.lowercase_ascii s in
         let s' =
           if Filename.check_suffix ls ".so" ||
             Filename.check_suffix ls ".dylib"
```

Comment:
Using `lowercase_ascii` breaks 4.02.3 compatibility. The fix is easy:
```suggestion
        let s' =
          if Filename.check_suffix s ".so" ||
            Filename.check_suffix s ".dylib"
```
`Filename.check_suffix` has done the correct thing since the dawn of time (https://github.com/ocaml/ocaml/commit/4ef227d589e2c01dc7fda4e774a6611f670c3ca7), even though it wasn't documented until the year before last (https://github.com/ocaml/ocaml/commit/a7a76fd4e8bd0d5ddce6857c0eed1f8652f720b6)